Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp

```diff
@@ -172,13 +172,10 @@ FloatPoint ScrollingTreeScrollingNode::maximumScrollPosition() const
 
 bool ScrollingTreeScrollingNode::eventCanScrollContents(const PlatformWheelEvent& wheelEvent) const
 {
-#if PLATFORM(GTK)
-    // In case of GTK platform the end of momentum scroll events
-    // always have a delta of 0. More information in the documentation
-    // of the API gdk_event_is_scroll_stop_event.
+    // The end of momentum scroll event can scroll contents even if with delta of 0.
+    // More information in the documentation of the API gdk_event_is_scroll_stop_event.
```

Comment:
It's weird to point to GTK documentation to explain a cross-platform change, In commit message you say this is only for ports using nicosia, but this is cross-platform, no?

Reply:
You are right. For some reason that I can't explain I was reading this code as Nicosia specific.

For the gdk_event_is_scroll_stop_event,  also you are right. Maybe @alexgcastro can do some suggestion for complementing the comment.

Reply:
Finally I found why this change is also valid for WPE. The reason why is in the  implementation of the  **libwpe** `handle_axis_event` API. In https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/API/wpe/WPEView.cpp#L189

```
        // We treat an axis motion event with a value of zero to be equivalent
        // to a 'stop' event. Motion events with zero motion don't exist naturally,
        // so this allows a backend to express 'stop' events without changing API.
        // The wheel event phase is adjusted accordingly.
        WebWheelEvent::Phase phase = WebWheelEvent::Phase::PhaseChanged;
        WebWheelEvent::Phase momentumPhase = WebWheelEvent::Phase::PhaseNone;
```



The motion events for the WPE port it comes through the UI-process using the `wpe_view_backend_dispatch_axis_event()` method of the **libwpe** API. For example, for Cog (@aperezdc):

```cpp
    struct wpe_input_axis_event event = {
        wpe_input_axis_event_type_motion,
        time,
        wl_data.pointer.x,
        wl_data.pointer.y,
        axis,
        - wl_fixed_to_int (value),
    };

    wpe_view_backend_dispatch_axis_event (wpe_view_data.backend, &event);
```

The  `wpe_view_backend_dispatch_axis_event()` method basically delegates the implementation in a backend what is implemented in `WebKit/UIProcess/API/wpe/WPEView.cpp` (WebKit):

``` cpp
void
wpe_view_backend_dispatch_axis_event(struct wpe_view_backend* backend, struct wpe_input_axis_event* event)
{
    if (backend->input_client)
        backend->input_client->handle_axis_event(backend->input_client_data, event);
}
```
in `src/view-backend.c`

... and finally, in https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/API/wpe/WPEView.cpp#L189 the motion event is interpreted artificially as an `phase::ended` event. (//CC @zdobersek )